### PR TITLE
Add support for folder decoupling

### PR DESCRIPTION
**Pull request by GiniAppsPartners**
### Brief Summary of Changes

Add support for folder decoupling

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

**Review by adimiz1 — APPROVED** [review]

File: Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift [hunk 1]
```diff
@@ -27,6 +27,8 @@ import XCTest
 
 class UploaderTests: NetworkBaseTest {
 
+    let allowFolderDecouplingCalls: Bool = false
```
Comment (adimiz1) [inline Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift:30]: Fix to pick from environment variable

File: Cloudinary/Classes/Core/Features/Helpers/Results/Helpers/CommonResultKeys.swift [hunk 2]
```diff
@@ -25,38 +25,40 @@
...
-        case .context:               return "context"
-        case .phash:                 return "phash"
-        case .info:                  return "info"
-        case .eager:                 return "eager"
-        case .accessControl:         return "access_control"
-        case .qualityAnalysis:       return "quality_analysis"
-        case .accessibilityAnalysis: return "accessibility_analysis"
-        case .coordinates:           return "coordinates"
-        case .originalFilename:      return "original_filename"
+        case .publicId:                 return "public_id"
```
Comment (adimiz1) [inline Cloudinary/Classes/Core/Features/Helpers/Results/Helpers/CommonResultKeys.swift:32]: I understand you added more variables to the description variable but the indentation, tabs and spaces should stay the same and they changed.

File: Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift (hunk 1 above)
Comment (adimiz1) [reply Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift:30]: Hi Alex, if you please change the `allowFolderDecouplingCalls` to be taken from environment variable and not a `Bool`

**Review by adimiz1 — CHANGES_REQUESTED** [review]
Added comments on relevant lines and classes.

File: Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift (hunk 1 above)
Comment (GiniAppsPartners) [reply Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift:30]: What environment variable specifically?

Comment (adimiz1) [reply Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift:30]: CLD_TEST_FOLDER_DECOUPLING
I would also suggest to look at a function called `shouldSkipTest` in `NetworkBaseTest`

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**